### PR TITLE
doc: use DX-friendly names in configuration option documentation

### PR DESCRIPTION
**Pull request by dbkinder**
Change the generated config option documentation to use the DX-friendly
names defined for the configurator UI (instead of the XML element name
hierarchy previously used).

Options are grouped by the top-level section (aka complex type) they
belong to and then sorted alphabetically with these groups.

Use badges to indicate where options can be found in the configurator UI
and whether they're applicable to the Hypervisor or Pre/Post/Service VM.
Add a custom css style for the config-option doc that puts the first
paragraph of a glossary item on the same line as the glossary term so
these badges look pretty.

Added a acrn-custom.js patch that copies the alt text for images into a
title property for images within the config-doc document.  This provides
tooltip text when hovering over the badges.

Don't display options not visible in the configurator UI (elements with
acrn:views="").

A missing acrn:views or acrn:applicable-vm means we look for an
applicable value from an ancestor element.

Add processing of a second xs:documentation element within an
xs:annotation element. This second documentation element's content will
be appended as a new paragraph to the first xs:documentation content in
the generated documentation. Only the first xs:documentation element is
used by the Configurator for its tooltips.

Update documents that were referring to options by their XML names.

Because we're now using a glossary to provide links to config options,
we can't duplicate option names or glosary names anywhere in the doc
set.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

**Comment by acrnsi** [comment]
Can one of the admins verify this patch?

**Comment by dbkinder** [comment]
A couple of duplicate "DX-friendly" names in the schema need to be resolved to fix the doc build error.

**Comment by dbkinder** [comment]
@amyreye @NanlinXie I removed the "on hold" as I'm ready for review and merge this to get us started with new config option documentation based on DX-friendly names.  We can continue to tweak as needed in additional PRs and with edits to the xsd file attributes and documentation content.

File: doc/scripts/configdoc.xsl [hunk 1]
```diff
@@ -1,165 +1,226 @@
...
-
     <xsl:apply-templates select="xs:complexType[@name='ACRNConfigType']">
       <xsl:with-param name="level" select="2"/>
       <xsl:with-param name="prefix" select="''"/>
+      <xsl:with-param name="parent" select="."/>
     </xsl:apply-templates>
-
   </xsl:template>
-
+  <!-- Walk througn all the ACRNConfigType element's children -->
```
Comment (amyreye) [inline doc/scripts/configdoc.xsl:17]:
```suggestion
  <!-- Walk through all the ACRNConfigType element's children -->
```

File: doc/scripts/configdoc.xsl [hunk 2]
```diff
@@ -1,165 +1,226 @@
...
+          <xsl:otherwise>
+            <!-- element doesn't have a named type, check for an unnamed simpleType child -->
+            <xsl:apply-templates select="descendant::xs:simpleType">
+              <xsl:with-param name="level" select="$level"/>
+              <xsl:with-param name="prefix" select="''"/>
+              <xsl:with-param name="parent" select="$parent"/>
+            </xsl:apply-templates>
+          </xsl:otherwise>
+        </xsl:choose>
+        <!-- Removed printing occurance information
```
Comment (amyreye) [inline doc/scripts/configdoc.xsl:187]:
```suggestion
        <!-- Removed printing occurrence information
```

File: doc/scripts/configdoc.xsl [hunk 3]
```diff
@@ -1,165 +1,226 @@
...
+          <xsl:otherwise>
+            <!-- element doesn't have a named type, check for an unnamed simpleType child -->
+            <xsl:apply-templates select="descendant::xs:simpleType">
+              <xsl:with-param name="level" select="$level"/>
+              <xsl:with-param name="prefix" select="''"/>
+              <xsl:with-param name="parent" select="$parent"/>
+            </xsl:apply-templates>
+          </xsl:otherwise>
+        </xsl:choose>
+        <!-- Removed printing occurrance information
```
Comment (amyreye) [inline doc/scripts/configdoc.xsl:187]:
```suggestion
        <!-- Removed printing occurrence information
```

**Review by NanlinXie — APPROVED** [review]

**Comment by dbkinder** [comment]
@NanlinXie @wenlingz CI seems stuck -- 2 hours and still running?

**Pull request merged**